src/VM/VM.cpp

```diff
@@ -171,6 +171,29 @@ void VM::pushDataPointer(void* value, unsigned int type)
     _dataStack.push(pointer);
 }
 
+std::string VM::popDataString()
+{
+    auto stackValue = _dataStack.pop();
+    if (stackValue->type() == VMStackValue::TYPE_POINTER)
+    {
+        auto stackPointerValue = dynamic_cast<VMStackPointerValue*>(stackValue);
```

Comment:
Above you check for `VMStackValue::TYPE_POINTER` and here you use `dynamic_cast`, but few lines below, in the same kind of situation, you use `static_cast`. If there is any danger the cast in line 179 could fail, can you add check against nullptr or otherwise change it to `static_cast`?